src/backtrace.h

```diff
@@ -0,0 +1,186 @@
+/* -*- C -*-
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
+ * DE-AC04-94AL85000 with Sandia Corporation, the U.S.  Government
+ * retains certain rights in this software.
+ *
+ * Copyright (c) 2018 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+#ifndef BACKTRACE_H
+#define BACKTRACE_H
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include "shmem_env.h"
+
+/* Backtrace through execinfo */
+#if defined(USE_BT_EXECINFO)
+#include <execinfo.h>
+/* Maximum total number of backtraces */
+#define MAX_BT_SIZE 1024
+
+static inline 
+void backtrace_execinfo(void) {
+    static void *btaddr[MAX_BT_SIZE];
+    size_t size, i;
+    char **fnnames = NULL;
+
+    size = backtrace(btaddr, MAX_BT_SIZE); 
+    fnnames = backtrace_symbols(btaddr, size);
```

Comment:
Good idea. Thanks.

Reply:
@davidozog Having a second thought here. With the current code, `USE_BT_EXECINFO` is defined only when the backtrace symbols are present. So, we do not really have to check `HAVE_BACKTRACE_SYMBOLS` since the whole routine is already checked with `USE_BT_EXECINFO`. 
I have added the check for NULL for `fnnames` though. Let me know if you feel this is ok.

Reply:
Ah, I see.  Maybe then this concern starts in the `configure.ac` [file here](https://github.com/Sandia-OpenSHMEM/SOS/pull/795/files#diff-67e997bcfdac55191033d57a16d1408aR438).

Should we be checking for `ac_cv_func_backtrace_symbols` in addition to `ac_cv_func_backtrace`?

Reply:
Thanks @davidozog. Yes, I added it now.